downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -0,0 +1,59 @@
+package org.reactome.release.downloadDirectory.GenerateGOAnnotationFile;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+
+import java.util.*;
+
+public class CellularComponentAnnotationBuilder {
+
+    private static final List<String> speciesWithAlternateGOCompartment = new ArrayList<>(Arrays.asList("11676", "211044", "1491", "1392"));
+    private static final List<String> microbialSpeciesToExclude = new ArrayList<>(Arrays.asList("813", "562", "491", "90371", "1280", "5811"));
+    private static final String PROTEIN_BINDING_ANNOTATION = "0005515";
+
+    public static void processCellularComponents(GKInstance reactionInst) throws Exception {
+        Set<GKInstance> proteinInstances = GOAGeneratorUtilities.retrieveProteins(reactionInst);
+        processProteins(proteinInstances, reactionInst);
```

Comment:
Maybe the body of `processProteins` could be put into this method since it only retrieving the proteins and then delegating to that method?

Reply:
Yea I had thought of that when creating it. My only concern was making it difficult for someone to follow the code flow. Is it a decided convention that embedding functions in arguments is OK? If this is one of those 'holy wars' I think I would fall on the side of not embedding, for readability's sake

Reply:
I _think_ that @jweiser 's point is that the public function `processCellularComponents` only has two lines: getting proteins and then passing them to `processProteins`, so in that case, the body of `processProteins` could be moved into `processCellularComponents` (but if I got it wrong, feel free to correct me). I don't think there's any "holy war" here. If you think it's more readable having two separate functions instead of one, I think that's OK.

Reply:
Yeah, I'm not adamant that they are combined, but I do think it might be a good idea.  For embedding functions in arguments, do you mean moving line 15 into the first argument position of line 16?  If so, that's not necessary.  I was just suggesting that the body of `processProteins` replace line 16 but line 15 can stay as it is.

Reply:
Done